### PR TITLE
test-validator: Try to reset deployment slot on all cloned accounts

### LINKED ISSUE #522: test-validator: Cloned program data accounts don't work as expected
### Problem

This is similar to https://github.com/anza-xyz/agave/issues/436 — which was fixed by https://github.com/anza-xyz/agave/pull/501 — however, the fix only includes programs that are cloned using the `--clone-upgradeable-program` option. If an account is cloned using `--clone` — which is what Anchor [does](https://github.com/coral-xyz/anchor/blob/c66b3232938186623c535cec8deba6b98353420b/cli/src/lib.rs#L3292-L3340) — the problem of not being able to interact with the programs still persist. Anchor can switch to use `--clone-upgradeable-program` but it will still be broken on all existing Anchor versions.

### Solution

Always set deployment slot of cloned program data accounts to `0`.

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

As pointed out in #522, the deployment slot on program data is only reset if the `--clone-upgradeable-program` flag is used. Previous setups using the plain `--clone` flag will still get a deployment slot in the future.

#### Summary of Changes

Although the better option is to move people to use `--clone-upgradeable-program`, the deployment slot addition was still a breaking change for downstream users, so try to reset the deployment slot on any cloned account, and ignore the error.

Fixes #522

### CODE REVIEW

File: test-validator/src/lib.rs [hunk 1]
```diff
@@ -348,7 +389,14 @@ impl TestValidatorGenesis {
...
             rpc_client,
             skip_missing,
-            |_address, account| Ok(AccountSharedData::from(account)),
+            |address, account| {
+                let account_shared_data = AccountSharedData::from(account);
+                // ignore the error
+                Ok(
+                    try_transform_program_data(address, account_shared_data.clone())
+                        .unwrap_or(account_shared_data),
+                )
```
Comment (buffalojoec) [inline test-validator/src/lib.rs]: Is this clone here avoidable? If `try_transform_program_data` took a mutable reference, couldn't you do this here?

```rust
let mut account_shared_data = AccountSharedData::from(account);
try_transform_program_data(&mut account_shared_data).ok(); // Error still ignored.
account_shared_data
```

And then in the other function:

```rust
let mut account_shared_data = AccountSharedData::from(account);
try_transform_program_data(&mut account_shared_data)?;
account_shared_data
```

**Review by buffalojoec — COMMENTED** [review]
Lgtm, just one comment about the cloning!

File: test-validator/src/lib.rs (hunk 1 above)
Comment (joncinque) [reply test-validator/src/lib.rs]: Ah yeah that's a great idea, done!

**Review by buffalojoec — APPROVED** [review]
Nice! Thanks for rolling this out.

**Review by jacobcreech — APPROVED** [review]
Thanks for addressing this so quickly!

**Pull request merged**